spritecontrol.cpp

```diff
@@ -119,6 +120,46 @@ void PatternPlayer::set_pattern(PatternName pattern) {
 	m_hash_offset++;
 }
 
+void PatternPlayer::handle_off_screen_sprites(bool wrap_sprites) {
```

Comment:
Rather than having a single function that does two dramatically different things based on a boolean parameter, consider splitting this into two functions and then deciding which one to call based on what you would haved pass to this function. 
```cpp
// SpriteChoreographer::update()
auto should_wrap = non_screen_wrapping_patterns.contains(m_pattern);
if (should_wrap) {
  m_current_player->wrap_off_screen_sprites();
} else {
  m_current_player->clamp_off_screen_sprites();
}
```
This spreads out responsibility more so that `handle_off_screen_sprites` isn't doing so many things. Plus, it always feels a little icky to have a function that takes a single boolean because it isn't clear from the call site what the difference between passing `true` and `false` would be or what that parameter actually means.

Reply:
nitpicking your nitpick: `should_wrap` should be the opposite here.

yeah, I get the bool parameter thing. the next best thing I can think of (while keeping it one function) is to use an enum instead, but that also feels rather unnecessary?